### DISCUSSION TITLE
.VxmDenseSemiSupervisedSeg perform better on CPU than GPU

### DISCUSSION THREAD

**Discussion by AJPan0118** [voxelmorph/voxelmorph#494] [Q&A]
hi there, I am trying to train the semi supervised segmentation model with some 3D prostate CT. The model perform pretty well when I trained the model on the CPU. However, with exactly same network architecture, parameters and everything, the model does not perform well on GPU. Anyone may know what may cause this?  Thank you.

**Comment by adalca** [comment]
Can you expand on what the specifics and differences are? what does 'does not perform well on GPU' mean -- training on GPU doesn't converge, it does but doesn't do quite as well, inference on GPU leads to an error, etc?

**Reply by AJPan0118** [reply]
hi thank you for your quick reply. When I trained the model on GPU, the loss did not reduce. I am expecting with exactly same parameters, the performance on GPU and CPU should be close. I did try different parameters on GPU as well. However, the loss fluctuate a lot on GPU.

**Comment by adalca** [comment]
I dont see why this would be, so seeing some graphs might help. What loss function are you using on the GPU?

**Reply by AJPan0118** [reply]
hi, I am using LCC on intensity, MSE on contour constraints and L2 for regularization. The loss reduces fine on CPU with exactly same loss and parameters.  I found this very odd too. Please see the attached graphs for training on CPU and GPU with exactly same parameters respectively. 
![image](https://user-images.githubusercontent.com/119630978/205540313-6d48688a-4fac-471c-9bcf-d0ba4f3b041c.png)

I have been trying different parameters for GPU training, however I am not able to get reasonable results yet.  I will keep you updated if I find the reason.
